cmd/helm/status.go

```diff
@@ -93,56 +94,77 @@ func (s *statusCmd) run() error {
 		return prettyError(err)
 	}
 
-	switch s.outfmt {
+	output, err := PrintStatusFormated(s.outfmt, res)
+
+	if err != nil {
+		return fmt.Errorf("Failed to Marshal YAML output: %s", err)
+	}
+
+	fmt.Fprintf(s.out, output)
+
+	return nil
+}
+
+// PrintStatusFormated prints out the status of a release. Shared because also used by
+// install / upgrade
+func PrintStatusFormated(format string, res *services.GetReleaseStatusResponse) (string, error) {
+	var output string
+	var err error
+
+	switch format {
 	case "":
-		PrintStatus(s.out, res)
-		return nil
+		output = printStatus(res)
+
 	case "json":
-		data, err := json.Marshal(res)
+		o, err := json.Marshal(res)
 		if err != nil {
-			return fmt.Errorf("Failed to Marshal JSON output: %s", err)
+			return "", fmt.Errorf("Failed to Marshal JSON output: %s", err)
 		}
-		s.out.Write(data)
-		return nil
+		output = string(o)
+
 	case "yaml":
-		data, err := yaml.Marshal(res)
+		o, err := yaml.Marshal(res)
 		if err != nil {
-			return fmt.Errorf("Failed to Marshal YAML output: %s", err)
+			return "", fmt.Errorf("Failed to Marshal YAML output: %s", err)
 		}
-		s.out.Write(data)
-		return nil
+		output = string(o)
+
+	default:
+		err = fmt.Errorf("Unknown output format %q", err)
 	}
 
-	return fmt.Errorf("Unknown output format %q", s.outfmt)
+	return output, err
 }
 
-// PrintStatus prints out the status of a release. Shared because also used by
-// install / upgrade
-func PrintStatus(out io.Writer, res *services.GetReleaseStatusResponse) {
+func printStatus(res *services.GetReleaseStatusResponse) string {
```

Comment:
Looking at this again, I just realized this is a breaking change. This function was publicly exported, so it will need to remain that way, though the underlying logic can be shifted around